app/moves/middleware.js

```diff
@@ -90,4 +94,34 @@ module.exports = {
       next(error)
     }
   },
+  setMovesByDateAllLocations: async (req, res, next) => {
+    const { moveDate } = res.locals
+
+    if (!moveDate) {
+      return next()
+    }
+
+    try {
+      const userLocations = get(req.session, 'user.locations', []).map(
+        loc => loc.id
+      )
+
+      if (userLocations.length === 0) {
+        userLocations.push(null)
```

Comment:
What's the purpose of this? Is this to handle when the user has no locations?

Reply:
Yes, that's correct. If the user hits view all, but there are no user locations configured.

Reply:
In theory, that shouldn't even happen as the global middleware will redirect the user to the locations sub-app if there are no locations so we can probably get rid of this and assume this will always be an array.

Reply:
I wonder if returning `next()` if there are no locations rather than pushing `null` into the array and running the rest of the function is neater if we feel we still need this here?